BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/distance/DistanceUtil.java

```diff
@@ -31,13 +31,19 @@ public static Location walkingSpeedFilter(Location oldLocation, Location newLoca
         return speedFilter(oldLocation, newLocation, HUMAN_WALKING_SPEED);
     }
 
-    public static Location speedFilter(Location oldLocation, Location newLocation, double metersPerSecond) {
+    /**
+     * Define a maximum movement speed to restrain a new location being further away than the
+     * distance possible in that time window.
+     */
+    public static Location speedFilter(Location oldLocation, Location newLocation, double maximumSpeed) {
         double distance = oldLocation.getDistanceTo(newLocation);
-        double angle = oldLocation.getAngleTo(newLocation);
         long timestampDelta = newLocation.getTimestamp() - oldLocation.getTimestamp();
-        metersPerSecond = metersPerSecond * ((float) timestampDelta / TimeUnit.SECONDS.toMillis(1));
-        if (distance > metersPerSecond) {
-            return oldLocation.getShiftedLocation(metersPerSecond, angle);
+        double currentSpeed = distance * ((float) timestampDelta / TimeUnit.SECONDS.toMillis(1));
```

Comment:
Should be `speed = distance / time`. If your tests passed, there's something wrong (or missing) with your tests, too.